### DISCUSSION TITLE
Save images as generic jog/png instead of base64

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#15428] [How to]
Facing issues with base64 conversion of images, and I would like to save all uploaded images in generic format not base64.
How to do it?

**Comment by s2b-git** [comment]
The reason I am asking is that I need to convert all images to certain format-size (thousands of them).

**Comment by mariuszkrzaczkowski** [comment]
there is no other ready-made solution at Wyswig.
you can use the field to attach photos
attaching files is insecure therefore it is complicated to do, but it is possible

**Comment by s2b-git** [comment]
marius what about base64 ? how to save images in source format without conversion?

**Reply by mariuszkrzaczkowski** [reply]
explain because I don't understand

**Comment by s2b-git** [comment]
CK Editor and image fields saving images in base64, here we see it pretty disturbing due to length of chars generated by such encoding.
We want to save all images embed in CK/fields in original format and uploaded to storage

Best

**Comment by leo719044** [comment]
I made it with ckfinder. you can try it.

**Reply by vovpff** [reply]
Could you provide guide how to emplement it?

**Reply by leo719044** [reply]
1. Download CKFinder and put it under /public_html
https://download.cksource.com/CKFinder/CKFinder%20for%20PHP/3.5.2/ckfinder_php_3.5.2.zip

2. Open /public_html/vendor/ckeditor/ckeditor/config.js and paste
    config.uploadUrl= '/ckfinder/core/connector/php/connector.php?command=QuickUpload&type=Files&responseType=json';
    config.filebrowserBrowseUrl= '/ckfinder/ckfinder.html';
    config.filebrowserImageBrowseUrl= '/ckfinder/ckfinder.html?type=Images';
    config.filebrowserUploadUrl= '/ckfinder/core/connector/php/connector.php?command=QuickUpload&type=Files';
    config.filebrowserImageUploadUrl= '/ckfinder/core/connector/php/connector.php?command=QuickUpload&type=Images';

**Reply by leo719044** [reply]
I set the root directory to /public_html

**Reply by mariuszkrzaczkowski** [reply]
You did it the wrong way, and you have a serious security gap. PHP files are not part of CRM, so anyone (without login) can freely work on all your files.

**Reply by leo719044** [reply]
I try to disable file browser feature in ckfinder, does it help ??

**Reply by mariuszkrzaczkowski** [reply]
no, all php code must be part of CRM without it, everyone will always be able to communicate in public

**Reply by vovpff** [reply]
@mariuszkrzaczkowski could you guide how to install ckfinder and connect it to ckeditor safe and secure?

**Reply by mariuszkrzaczkowski** [reply]
it cannot be done easily, the whole mechanism has to be written, with support for access to uploaded files

**Reply by vovpff** [reply]
Oh shit. Ok. This is sad

**Reply by leo719044** [reply]
actually, I made it without CKfinder using self definded upload.php file. 

but the problem is I can not paste image to input box directly.

**Comment by s2b-git** [comment]
@leo719044 what do you mean: "you can try it"?
Do you have solution ready?

**Reply by leo719044** [reply]
Yes, please try the steps above.
